packages/vendure-plugin-qls-fulfillment/src/qls-plugin.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  EventBus,
+  PluginCommonModule,
+  Type,
+  VendurePlugin,
+} from '@vendure/core';
+import { adminApiExtensions } from './api/api-extensions';
+import { fullSyncPermission } from './config/permissions';
+import { QlsAdminResolver } from './api/qls-admin.resolver';
+import { QlsWebhooksController } from './api/qls-webhooks-controller';
+import { PLUGIN_INIT_OPTIONS } from './constants';
+import { QlsOrderService } from './services/qls-order.service';
+import { QlsProductService } from './services/qls-product.service';
+import { QlsPluginOptions } from './types';
+import { AdminUiExtension } from '@vendure/ui-devkit/compiler';
+import path from 'path';
+import { variantCustomFields } from './custom-fields';
+
+@VendurePlugin({
+  imports: [PluginCommonModule],
+  providers: [
+    {
+      provide: PLUGIN_INIT_OPTIONS,
+      useFactory: () => QlsPlugin.options,
+    },
+    QlsProductService,
+    QlsOrderService,
+  ],
+  controllers: [QlsWebhooksController],
+  configuration: (config) => {
+    config.authOptions.customPermissions.push(fullSyncPermission);
+    config.customFields.ProductVariant.push(...variantCustomFields);
+
+    return config;
+  },
+  compatibility: '>=3.2.0',
+  adminApiExtensions: {
+    schema: adminApiExtensions,
+    resolvers: [QlsAdminResolver],
+  },
+})
+export class QlsPlugin {
+  static options: QlsPluginOptions;
+
+  constructor(
+    private eventBus: EventBus,
+    private qlsService: QlsProductService
+  ) {}
```

Comment:
The injected dependencies are not used.

I personally prefer registering event listeners in the plugin (instead of in the services) and let services only perform the actual work to be done. This would allow having everything event related in one place and (if wanted) use a shared job queue for all jobs, which are related to this plugin.

But I guess, that's a matter o taste 🙂

Reply:
I removed the unused dependencies.

Valid points, I don't really have an opinion on that. Currently, the QLS order service listens for events that relate to orders, and QLS product service what relates to products. The path of least resistance is to leave it like that now :laughing: